app/assets/javascripts/oxalis/geometries/materials/plane_material_factory.js

```diff
@@ -603,19 +605,13 @@ float linearizeVec3ToIndexWithMod(vec3 position, float base, float m) {
 
 // Similar to the transDim function in dimensions.js, this function transposes dimensions for the current plane.
 vec3 transDim(vec3 array) {
-  <%= (function () {
-      switch (planeID) {
-        case OrthoViews.PLANE_XY:
-          return "return array;";
-        case OrthoViews.PLANE_YZ:
-          return "return vec3(array.z, array.y, array.x);"; // [2, 1, 0]
-        case OrthoViews.PLANE_XZ:
-          return "return vec3(array.x, array.z, array.y);"; //[0, 2, 1]"
-        default:
-          throw new Error("Invalid planeID provided to fragment shader");
-      }
-    })()
-  %>
+  if (planeID == <%= OrthoViewIndices.PLANE_XY %>) {
+    return array;
+  } else if (planeID == <%= OrthoViewIndices.PLANE_YZ %>) {
+    return vec3(array.z, array.y, array.x); // [2, 1, 0]
```

Comment:
While you are at it: I think `return array.zyx` and `return array.xzy` should also work :)